ambassador/tests/t_mappingtests.py

```diff
@@ -420,3 +412,107 @@ def check(self):
                 assert r.json['headers']['Foo'] == 'FooF'
                 assert 'Zoo' not in r.json['headers']
                 assert 'Aoo' not in r.json['headers']
+
+class AddReqHeadersMapping(MappingTest):
+    parent: AmbassadorTest
+    target: ServiceType
+
+    @classmethod
+    def variants(cls):
+        for st in variants(ServiceType):
+            yield cls(st, name="{self.target.name}")
+
+    def config(self):
+        yield self, self.format("""
+---
+apiVersion: ambassador/v1
+kind:  Mapping
+name:  {self.name}
+prefix: /{self.name}/
+service: http://{self.target.path.fqdn}
+add_request_headers:
+    zoo:
+        append: False
+        value: Zoo
+    aoo:
+        append: True
+        value: aoo
+    boo:
+        value: boo
+    foo: Foo
+""")
+
+    def queries(self):
+        yield Query(self.parent.url(self.name + "/"), headers={
+            "zoo": "ZooZ",
+            "aoo": "AooA",
+            "boo": "BooB",
+            "foo": "FooF"
+        })
+
+    def check(self):
+        for r in self.results:
+            if r.backend:
+                assert r.backend.request.headers['zoo'] == ['Zoo']
+                assert r.backend.request.headers['aoo'] == ['AooA','aoo']
+                assert r.backend.request.headers['boo'] == ['BooB','boo']
+                assert r.backend.request.headers['foo'] == ['FooF','Foo']
+
+
+class LinkerdHeaderMapping(AmbassadorTest):
+    target: ServiceType
+
+    def init(self):
+        self.target = HTTP()
+        self.target_no_header = HTTP(name="noheader")
+
+    def config(self):
+        yield self, self.format("""
+
+---
+apiVersion: ambassador/v0
+kind:  Module
+name:  ambassador
+config:
+  add_linkerd_headers: true
+---
+apiVersion: ambassador/v1
+kind: Mapping
+name: {self.target_no_header.path.k8s}
+prefix: /target_no_header/
+service: {self.target_no_header.path.fqdn}
+add_linkerd_headers: false
+add_request_headers:
```

Comment:
I feel like we need a test that has `add_linkerd_headers` _without_ `add_request_headers` -- that should be easy to add, right?